kafka-rest/src/main/java/io/confluent/kafkarest/ratelimit/RequestRateLimiterProduceBytesFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.ratelimit;
+
+import io.confluent.kafkarest.config.ConfigModule;
+import java.time.Duration;
+import javax.inject.Inject;
+
+public class RequestRateLimiterProduceBytesFactory extends RequestRateLimiterFactory {
+
+  RequestRateLimiterProduceBytes requestRateLimiterProduceBytes;
+
+  @Inject
+  public RequestRateLimiterProduceBytesFactory(
+      RateLimitBackend backend,
+      @ConfigModule.ProduceRateLimitBytesConfig Integer permitsPerSecond,
+      @ConfigModule.RateLimitTimeoutConfig Duration timeout) {
+    super(backend, permitsPerSecond, timeout);
+    requestRateLimiterProduceBytes = new RequestRateLimiterProduceBytes(super.provide());
```

Comment:
Don't cache the provided instance. Make sure to return a new instance everytime. HK2 will take care of the lifetime of the instance.